File: CovidWatch/app/src/main/java/com/riskre/covidwatch/ContactEventsAdapter.java
package com.riskre.covidwatch;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class ContactEventsAdapter extends RecyclerView.Adapter<ContactEventsAdapter.ViewHolder> {

    private static final String TAG = "ContactEventsAdapter";
    private ArrayList<String> contact_logs;
    private Context mContext;

    public ContactEventsAdapter(ArrayList<String> contact_logs, Context mContext) {
        this.contact_logs = contact_logs;
        this.mContext = mContext;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_listitem, parent, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Log.i(TAG, "onBindViewHolder: called");
        holder.data.setText(contact_logs.get(position));
    }

    @Override
    public int getItemCount() {
        return contact_logs.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        TextView data;
        RelativeLayout parentLayout;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            data = itemView.findViewById(R.id.data);
            parentLayout = itemView.findViewById(R.id.parent_layout);
        }
    }
}

File: CovidWatch/app/src/main/java/org/covidwatch/android/ui/contactevents/ContactEventsAdapter.kt
package org.covidwatch.android.ui.contactevents

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.paging.PagedListAdapter
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import org.covidwatch.android.R
import org.covidwatch.android.data.ContactEvent
import org.covidwatch.android.databinding.ListItemContactEventBinding


class ContactEventsAdapter() :
    PagedListAdapter<ContactEvent, ContactEventsAdapter.ContactEventViewHolder>(
        DIFF_CALLBACK
    ) {

    override fun getItemViewType(position: Int): Int {
        return R.layout.list_item_contact_event
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactEventViewHolder {
        return ContactEventViewHolder(
            DataBindingUtil.inflate(
                LayoutInflater.from(parent.context),
                viewType, parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: ContactEventViewHolder, position: Int) {
        val contactEvent: ContactEvent = getItem(position) ?: return
        holder.bind(contactEvent)
    }

    class ContactEventViewHolder(
        private val binding: ListItemContactEventBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(item: ContactEvent) {
            binding.apply {
                contactEvent = item
                executePendingBindings()
            }
        }

    }

    companion object {
        private val DIFF_CALLBACK = object :
            DiffUtil.ItemCallback<ContactEvent>() {
            // Contact event details may have changed if reloaded from the database,
            // but ID is fixed.
            override fun areItemsTheSame(
                oldConctactEvent: ContactEvent,
                newContactEvent: ContactEvent
            ) = oldConctactEvent.identifier == newContactEvent.identifier

            @SuppressLint("DiffUtilEquals")
            override fun areContentsTheSame(
                oldContactEvent: ContactEvent,
                newContactEvent: ContactEvent
            ) = oldContactEvent == newContactEvent
        }
    }

}
